rados/rados.go

```diff
@@ -96,17 +107,28 @@ func (c *RadosConn) MonCommand(args []byte) (buffer []byte, info string, err err
 
 	conn, err := c.newRadosConn()
 	if err != nil {
-		return nil, "", err
+		return
 	}
-	defer conn.Shutdown()
 
+	defer conn.Shutdown()
 	ll = ll.WithField("conn", conn.GetInstanceID())
 
-	ll.Trace("start executing mon command")
-
-	buffer, info, err = conn.MonCommand(args)
-
-	ll.WithError(err).Trace("complete executing mon command")
+	ch := make(chan bool, 1)
```

Comment:
Here and below, I think what you want is the signal channel pattern: https://medium.com/@matryer/golang-advent-calendar-day-two-starting-and-stopping-things-with-a-signal-channel-f5048161018

Reply:
-resolving due to earlier code change- 🙂 but thanks for linking that, I forgot there's already a pattern for this type of thing in Go